### DISCUSSION TITLE
Problema na ordem de operações matemáticas

### DISCUSSION THREAD

**Discussion by ivanmortimer** [dgadelha/Portugol-Webstudio#197] [Ideias e sugestões]
Olá,

Gostaria de informar que há uma incongruência entre o comentário sobre resultado esperado de uma expressão aritmética e o resultado de fato obtido no Exemplo "Prioridades", dentro da categoria "Operações Aritméticas". No comentário sobre as seguintes expressões:

```portugol
resultado = 1.0 + 2.0 / 3.0 * 4.0 
escreva("\nOperação: 1 + 2 / 3 * 4 = ", resultado)
```

é dito que:

```portugol
/*
 * Neste exemplo, a operação de divisão (/) será executada primeiro, 
 * seguida pela operação de multiplicação (*). Por último, será 
 * executada a operação de soma (+)
*/
```

Porém, fica claro pelo resultado que aparece na saída ("Operação: 1 + 2 / 3 * 4 = 1.1666666666666667"), que é a operação de MULTIPLICAÇÃO que é realizada primeiro, seguida da operação de DIVISÃO, seguida finalmente pela operação de SOMA. Se a ordem fosse a quela descrita no comentário acima (isto é, DIVISÃO, seguida de MULTIPLICAÇÃO, seguida de SOMA), o resultado obtido seria "Operação: 1 + (2 / 3) * 4 = 3.6666666666666665".

Atenciosamente,

Ivan.

**Comment by ivanmortimer** [comment]
Olá novamente,

Ao se analisar o código JavaScript correndo ao fundo, fica evidente qual é o problema:

    runtime.assign([
      scope.variables["resultado"],
      runtime.mathOperation("+", [
        new PortugolVar("real", 1.0),
        runtime.mathOperation("/", [
          new PortugolVar("real", 2.0),
          runtime.mathOperation("*", [
            new PortugolVar("real", 3.0),
            new PortugolVar("real", 4.0),
          ]),
        ]),
      ]),
    ]);

A operação de escopo mais interno (e, portanto, aquela que é executada primeiro) é a de MULTIPLICAÇÃO ("*"), o escopo imediatamente exterior a esse está associado à operação de DIVISÃO ("/"), e, por fim, o escopo mais exterior está associado à operação de ADIÇÃO ("+").

Atenciosamente,

Ivan.

**Reply by dgadelha** [reply]
Obrigado pelo report! Criei a issue #208 e estou fechando a discussão já que se trata de um bug. Vou atualizar lá quando conseguir chegar numa correção para o problema.
